src/components/data-management/SamplesTable.jsx

```diff
@@ -376,3 +396,7 @@ SamplesTable.propTypes = {
 };
 
 export default React.memo(SamplesTable);
```

Comment:
Out of curiosity, do you know why there is a `React.memo` here?

Reply:
No idea. I think it was just an effort to improve the `SamplesTable` performance. I don't think it's useful as `React.memo` only check for `prop` changes, not changes due to `useEffect` or `useSelector` (https://reactjs.org/docs/react-api.html#reactmemo). I think this can be removed?

Reply:
I do know that there's some rerendering happening in the parent component of `SamplesTable` (it's what I was working on in the upload improvement PR), maybe this is some kind of patch for that.

Reply:
I would keep it for now